Migrate to `module` script plugin

alexander-yevsyukov opened the pull request:
This PR:
  * Migrates module configuration to use `module` script plugin.
  * Extracts smoke JavaScript code generation into a module `time-js`.
  * Updates `config`.
  * Migrates to Kotest assertions.

alexander-yevsyukov committed:
Update config

alexander-yevsyukov committed:
Update config

alexander-yevsyukov committed:
Remove unused import

alexander-yevsyukov committed:
Bump Protobuf and plugin

alexander-yevsyukov committed:
Move JS code gen into a separate module

alexander-yevsyukov committed:
Optimise imports

alexander-yevsyukov committed:
Update dependency reports

alexander-yevsyukov committed:
Rename extension files to new convention

Also:
 * Migrate assertions to Kotest

alexander-yevsyukov committed:
Extract `module` script plugin

alexander-yevsyukov committed:
Address `gradle-doctor` error

alexander-yevsyukov committed:
Update build time

alexander-yevsyukov committed:
Improve readme language

alexander-yevsyukov committed:
Bump Kotlin -> `1.8.10`

alexander-yevsyukov committed:
Update dependencies

alexander-yevsyukov committed:
Improve type name

alexander-yevsyukov committed:
Migrate the remaining assertion

alexander-yevsyukov committed:
Add additional clean-up before `compileKotlin`

... to overcome GitHub build failure under Ubuntu which reports the error:

```
> Cannot fingerprint input file property 'sources': Failed to normalize content of '/home/runner/work/time/time/time/build/generated/source/proto/main/kotlin/io/spine/time/LocalDateChangeKt.kt'.
```

alexander-yevsyukov committed:
Update build time

alexander-yevsyukov committed:
Wait while the interim. dir exists

alexander-yevsyukov committed:
Improve `KotlinCompile` filtering

alexander-yevsyukov committed:
Disable Kotlin compile caching

alexander-yevsyukov committed:
Improved filtering

alexander-yevsyukov committed:
Filter more

alexander-yevsyukov committed:
Remove redundant wait loop

alexander-yevsyukov committed:
Turn off Gradle cache

armiol commented on time-js/README.md:
```diff
@@ -0,0 +1,11 @@
+# `time-js` module
+
+The purpose of this module is to be a smoke-test for Protobuf code generation into JavaScript.<br/>
```
Please kill this `<br/>`.

alexander-yevsyukov replied:
As discussed vocally, we need this break to make the paragraph easier to read. There's no seem to be no better way for formatting in Markdown, unless we want to have a separate paragraph. But here we don't.

armiol reviewed (approved):
@alexander-yevsyukov please see my comment regarding `time-js/README.md`.

Other than that, LGTM.

alexander-yevsyukov committed:
Restore Gradle cache

alexander-yevsyukov committed:
Use in-process Kotlin compile

alexander-yevsyukov committed:
Bump `base`

alexander-yevsyukov committed:
Remove `in-process` property for Kotlin execution

alexander-yevsyukov committed:
Apply new `buildSrc`

alexander-yevsyukov committed:
Apply new `buildSrc`

alexander-yevsyukov committed:
Optimise imports

alexander-yevsyukov committed:
Update build time

alexander-yevsyukov committed:
Update `config`

alexander-yevsyukov committed:
Remove outdated script

alexander-yevsyukov committed:
Bump McJava -> `2.0.0-SNAPSHOT.147`

alexander-yevsyukov committed:
Remove redundant empty line

alexander-yevsyukov committed:
Remove redundant script

alexander-yevsyukov committed:
Remove redundant renderer

There's no need to put `PrintBeforePrimaryDeclaration` because suppression plugin is no longer applied. Our custom suppression is no longer needed because Protobuf generates it itself.

alexander-yevsyukov committed:
Remove redundant renderer

There's no need to put `PrintBeforePrimaryDeclaration` because suppression plugin is no longer applied. Our custom suppression is no longer needed because Protobuf generates it itself.
